src/spatio_temporal_voxel_layer.cpp

```diff
@@ -342,14 +349,19 @@ void SpatioTemporalVoxelLayer::LaserScanCallback(
   const std::shared_ptr<buffer::MeasurementBuffer> & buffer)
 /*****************************************************************************/
 {
+  auto node = node_.lock();
```

Comment:
Seems like we should store the clock / logger in the layer as well so you don't need to clock the node on callbacks just to log / get a time.

Reply:
I am not sure I get a full understanding yet of the clock and logger mechanisms in ROS2, but here, could we use the inherited `clock_` and `logger_` from the Layer class instead ?
```
  rclcpp::Clock::SharedPtr clock_;
  rclcpp::Logger logger_{rclcpp::get_logger("nav2_costmap_2d")};
```
https://github.com/ros-planning/navigation2/blob/d3bf4ff895e47d79be2cab12bac0c1a45027c93f/nav2_costmap_2d/include/nav2_costmap_2d/layer.hpp#L142-L143

Reply:
That appears to be true - forgot about that, then you shouldn't need to 

```

  auto node = node_.lock();
  if (!node) {
    throw std::runtime_error{"Failed to lock node"};
  }
```

In 9 functions just to get loggers / clocks. I'm trying to remove those lines. Every time you `lock` a node, that is a nontrivial expense just to get a clock/logger that can be stored on initialization and accessed freely